src/pipecat/transports/base_input.py

```diff
@@ -211,10 +234,23 @@ async def _audio_task_handler(self):
 
             # Check VAD and push event if necessary. We just care about
             # changes from QUIET to SPEAKING and vice versa.
+            previous_vad_state = vad_state
             if self._params.vad_enabled:
                 vad_state = await self._handle_vad(frame, vad_state)
                 audio_passthrough = self._params.vad_audio_passthrough
 
+            if self._params.end_of_turn_analyzer:
+                is_speech = vad_state == VADState.SPEAKING or vad_state == VADState.STARTING
+                # If silence exceeds threshold, we are going to receive EndOfTurnState.COMPLETE
+                end_of_turn_state = self._params.end_of_turn_analyzer.append_audio(
+                    frame.audio, is_speech
+                )
+                if end_of_turn_state == EndOfTurnState.COMPLETE:
+                    await self._handle_end_of_turn_complete(end_of_turn_state)
+                # Otherwise we are going to trigger to check if the turn is completed based on the VAD
+                elif vad_state == VADState.QUIET and vad_state != previous_vad_state:
+                    await self._handle_end_of_turn()
+
```

Comment:
Can we move this to a function to keep this function smaller:

```
if self._params.end_of_turn_analyzer:
    await self._run_turn_analyzer(frame, vad_state, previous_vad_state)
```

Reply:
Done.